schema-registry/rest-service/src/main/java/com/hortonworks/registries/schemaregistry/webservice/SchemaRegistryResource.java

```diff
@@ -983,4 +1019,79 @@ public Response deleteSchemaVersion(@ApiParam(value = "Schema name", required =
             return WSUtils.respond(Response.Status.INTERNAL_SERVER_ERROR, CatalogResponse.ResponseMessage.EXCEPTION, ex.getMessage());
         }
     }
+
+    @GET
+    @Path("/branches")
+    @ApiOperation(value = "Get list of registered schema branches",
+            response = SchemaBranch.class, responseContainer = "List",
+            tags = OPERATION_GROUP_OTHER)
+    @Timed
+    @UnitOfWork
+    public Response getAllBranches(@Context UriInfo uriInfo) {
+        try {
+            Collection<SchemaBranch> schemaBranches = schemaRegistry.getAllBranches();
+            return WSUtils.respondEntities(schemaBranches, Response.Status.OK);
+        } catch (Exception ex) {
+            LOG.error("Encountered error while listing schema branches", ex);
+            return WSUtils.respond(Response.Status.INTERNAL_SERVER_ERROR, CatalogResponse.ResponseMessage.EXCEPTION, ex.getMessage());
+        }
+    }
+
+    @POST
+    @Path("/branch/version/{version}/create")
```

Comment:
It may be good to give api to add branch for each schema metadata like below and we should not really need create as POST refer that.
one possible way may be "/schemas/versionsById/{id}/branch".

Reply:
Done